ovs/src/main/java/org/dcsa/conformance/standards/ovs/checks/OvsChecks.java

```diff
@@ -30,124 +33,70 @@ public static ActionCheck responseContentChecks(
-                validateParameter(
-                    body,
-                    sspSupplier,
-                    OvsFilterParameter.VESSEL_NAME,
-                    "*/vesselSchedules/*/vesselName")));
-
-    checks.add(
-        JsonAttribute.customValidator(
-            "Validate carrierVoyageNumber exists and "
-                + "matches in JSON response if request parameter has carrierVoyageNumber",
-            body ->
-                validateParameter(
-                    body,
-                    sspSupplier,
-                    OvsFilterParameter.CARRIER_VOYAGE_NUMBER,
-                    "*/vesselSchedules/*/transportCalls/*/carrierExportVoyageNumber",
-                    "*/vesselSchedules/*/transportCalls/*/carrierImportVoyageNumber")));
-
-    checks.add(
-        JsonAttribute.customValidator(
-            "Validate universalVoyageReference exists and"
-                + " matches in JSON response if request parameter has universalVoyageReference",
-            body ->
-                validateParameter(
-                    body,
-                    sspSupplier,
-                    OvsFilterParameter.UNIVERSAL_VOYAGE_REFERENCE,
-                    "*/vesselSchedules/*/transportCalls/*/universalImportVoyageReference",
-                    "*/vesselSchedules/*/transportCalls/*/universalExportVoyageReference")));
-
-    checks.add(
-        JsonAttribute.customValidator(
-            "Validate UNLocationCode exists and "
-                + "matches in JSON response if request parameter has UNLocationCode",
-            body ->
-                validateParameter(
-                    body,
-                    sspSupplier,
-                    OvsFilterParameter.UN_LOCATION_CODE,
-                    "*/vesselSchedules/*/transportCalls/*/location/UNLocationCode")));
+            "Every response received during a conformance test must contain schedules",
+            body -> {
+              Set<String> validationErrors = new LinkedHashSet<>();
+              checkServiceSchedulesExist(body)
+                  .forEach(
+                      validationError ->
+                          validationErrors.add(
+                              "CheckServiceSchedules failed: %s".formatted(validationError)));
+              if (validationErrors.isEmpty()) {
+                return Set.of();
+              }
+              return validationErrors;
+            }));
 
     checks.add(
         JsonAttribute.customValidator(
-            "Validate facilitySMDGCode exists and "
-                + "matches in JSON response if request parameter has facilitySMDGCode",
-            body ->
-                validateParameter(
-                    body,
-                    sspSupplier,
-                    OvsFilterParameter.FACILITY_SMDG_CODE,
-                    "*/vesselSchedules/*/transportCalls/*/location/facilitySMDGCode")));
+            "If present, at least schedule attribute must match the corresponding query parameters",
```

Comment:
Does this mean at least one schedule? I didn't get it from this message string

Reply:
it should have been "at least one schedule". it corrrect it